File: mods/java/common/src/main/java/com/alaydriem/bedrockvoicechat/dto/Payload.java
package com.alaydriem.bedrockvoicechat.dto;

import lombok.Getter;

import java.util.List;

/**
 * Payload sent to the BVC server with player position data.
 */
@Getter
public class Payload {
    private final String game;
    private final List<PlayerData> players;

    public Payload(GameType gameType, List<PlayerData> players) {
        this.game = gameType.getValue();
        this.players = players;
    }
}

File: mods/java/common/src/main/kotlin/com/alaydriem/bedrockvoicechat/dto/Payload.kt
package com.alaydriem.bedrockvoicechat.dto

/**
 * Payload sent to the BVC server with player position data.
 */
data class Payload(
    val game: String,
    val players: List<PlayerData>
) {
    constructor(gameType: GameType, players: List<PlayerData>) : this(
        game = gameType.value,
        players = players
    )
}
